dao-voting/tests/harness.rs

```diff
@@ -0,0 +1,897 @@
+
+            let proposal = user.dao_voting.proposal(0).call().await.unwrap().value;
+
+            assert_eq!(
+                proposal,
+                ProposalInfo {
+                    author: Identity::Address(user.wallet.address()),
+                    yes_votes: asset_amount / 4,
+                    no_votes: asset_amount / 4,
+                    acceptance_percentage: 10,
+                    proposal_transaction: call_data,
+                    deadline: 15,
+                }
+            );
+        }
+    }
+
+    mod revert {
+        use super::*;
+
+        #[tokio::test]
+        #[should_panic]
+        async fn panics_on_not_enough_votes() {
+            let (_gov_token, gov_token_id, deployer, user, asset_amount) = setup().await;
+            deployer
+                .dao_voting
+                .constructor(gov_token_id)
+                .call()
+                .await
+                .unwrap()
+                .value;
+
+            let call_data = get_call_data(gov_token_id);
+
+            user.dao_voting
+                .create_proposal(10, 10, call_data)
+                .call()
+                .await
+                .unwrap()
+                .value;
+
+            user.dao_voting
+                .vote(true, 0, asset_amount / 4)
+                .call()
+                .await
+                .unwrap()
+                .value;
+        }
+
+        #[tokio::test]
+        #[should_panic]
+        async fn panics_on_expired_proposal() {
+            let (_gov_token, gov_token_id, deployer, user, asset_amount) = setup().await;
+            deployer
+                .dao_voting
+                .constructor(gov_token_id)
+                .call()
+                .await
+                .unwrap()
+                .value;
+
+            assert!(
+                deployer
+                    .gov_token
+                    .unwrap()
+                    .mint_and_send_to_address(100, user.wallet.address())
+                    .append_variable_outputs(1)
+                    .call()
+                    .await
+                    .unwrap()
+                    .value
+            );
+
+            let call_data = get_call_data(gov_token_id);
+
+            user.dao_voting
+                .create_proposal(1, 1, call_data)
+                .call()
+                .await
+                .unwrap()
+                .value;
+
+            let tx_params = TxParameters::new(None, Some(1_000_000), None, None);
+            let call_params = CallParameters::new(
+                Some(asset_amount),
+                Some(AssetId::from(*gov_token_id)),
+                Some(100_000),
+            );
+            user.dao_voting
+                .deposit()
+                .tx_params(tx_params)
+                .call_params(call_params)
+                .call()
+                .await
+                .unwrap()
+                .value;
+
+            user.dao_voting
+                .vote(true, 0, asset_amount / 4)
```

Comment:
again, magic division / numbers are confusing and not needed here. just use the full amount